File: src/main/java/io/bindernews/thegrackle/actions/SoundAction.java
package io.bindernews.thegrackle.actions;

import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.core.CardCrawlGame;

public class SoundAction extends AbstractGameAction {
    private final String sfx;

    public SoundAction(String sound) {
        actionType = ActionType.CARD_MANIPULATION;
        duration = DEFAULT_DURATION;
        sfx = sound;
    }

    @Override
    public void update() {
        CardCrawlGame.sound.play(sfx);
        isDone = true;
    }
}

File: src/main/kotlin/net/bindernews/grackle/actions/SoundAction.kt
package net.bindernews.grackle.actions

import com.megacrit.cardcrawl.actions.AbstractGameAction
import com.megacrit.cardcrawl.core.CardCrawlGame

class SoundAction(sound: String) : AbstractGameAction() {
    private val sfx: String

    init {
        actionType = ActionType.CARD_MANIPULATION
        duration = DEFAULT_DURATION
        sfx = sound
    }

    override fun update() {
        CardCrawlGame.sound.play(sfx)
        isDone = true
    }
}